### PR TITLE
Consolidating Sweep functionality

### PR DESCRIPTION
**Pull request by Tang8330**
## Context

With https://github.com/artie-labs/transfer/pull/360, we are now creating temporary tables with `unix ts` as a suffix. This PR is now building on top of it by consolidating the sweeping logic and removing the reliance on parsing comments which was to maintain backwards compatibility.

## Changes

* Changing the shared sweep function to retrieve the schema name from the query as opposed to from the `dbAndSchemaPair`.

^ We need to do this to support MSSQL as `dbo` is an alias for `public`

* Stood up a shared function for Sweep

### CODE REVIEW

File: clients/redshift/redshift.go [hunk 1]
```diff
@@ -41,13 +42,13 @@ func (s *Store) Label() constants.DestinationKind {
...
 }
 
-const (
-	describeNameCol        = "column_name"
-	describeTypeCol        = "data_type"
-	describeDescriptionCol = "description"
-)
-
 func (s *Store) GetTableConfig(tableData *optimization.TableData) (*types.DwhTableConfig, error) {
+	const (
```
Comment (Tang8330) [inline clients/redshift/redshift.go:46]: Just moved this inline since these constants are function specific.

File: clients/shared/sweep.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+		for rows != nil && rows.Next() {
+			var tableSchema, tableName string
+			if err != nil {
+				err = rows.Scan(&tableSchema, &tableName)
+				return err
+			}
+
+			if ddl.ShouldDeleteFromName(tableName) {
+				err = ddl.DropTemporaryTable(dwh, fmt.Sprintf("%s.%s.%s", dbAndSchemaPair.Database, tableSchema, tableName), true)
```
Comment (Tang8330) [inline clients/shared/sweep.go:33]: Note: Using `tableSchema` from the query result as opposed to `dbAndSchemaPair.Schema`

File: clients/snowflake/snowflake.go [hunk 3]
```diff
@@ -51,6 +52,25 @@ func (s *Store) GetTableConfig(tableData *optimization.TableData) (*types.DwhTab
...
+	}
+
+	queryFunc := func(dbAndSchemaPair kafkalib.DatabaseSchemaPair) (string, []any) {
+		return fmt.Sprintf(`
+SELECT 
+    table_schema, table_name
+FROM
+    %s.information_schema.tables
+WHERE
+    table_schema = UPPER(?) AND table_name ILIKE ?`, dbAndSchemaPair.Database), []any{dbAndSchemaPair.Schema, "%" + constants.ArtiePrefix + "%"}
```
Comment (nathan-artie) [inline clients/snowflake/snowflake.go:68]: Is it ever possible to have a `table_schema` that is not upper case? Should we do `UPPER(table_schema)`?

**Review by nathan-artie — APPROVED** [review]

File: clients/snowflake/snowflake.go (hunk 3 above)
Comment (Tang8330) [reply clients/snowflake/snowflake.go:68]: it is if the schema is escaped, but also no harm in adding UPPER(table_schema)

**Comment by Tang8330** [comment]
This is currently on hold, we will merge this later this week

**Pull request merged**